app/domain/authentication/authn_azure/validate_application_identity.rb

```diff
@@ -0,0 +1,102 @@
+require 'command_class'
+
+module Authentication
+  module AuthnAzure
+
+    Log = LogMessages::Authentication
+    Err = Errors::Authentication::AuthnAzure
+    # Possible Errors Raised: RoleNotFound, InvalidApplicationIdentity
+
+    ValidateApplicationIdentity = CommandClass.new(
+        dependencies: {
+            role_class:                 Role,
+            resource_class:             Resource,
+            application_identity_class: ApplicationIdentity,
+            logger:                     Rails.logger
+        },
+        inputs: %i(account service_id username xms_mirid oid)
+    ) do
+
+      def call
+        parse_xms_mirid
+        # compare xms_mirid with what is defined in annotations
+        validate_application_identity
+      end
+
+      private
+
+      # validate integrity of annotations
+      def application_identity
+        @application_identity ||= @application_identity_class.new(
+            role_annotations: role_annotations,
+            service_id: @service_id
+        )
+      end
+
+      def role_annotations
+        role.annotations
+      end
+
+      # convert host ID to full host ID, cucumber:host:somepolicy
+      # TODO For a Conjur user, an incorrect full id will be returned
+      def role_id
+        @role_id ||= @role_class.roleid_from_username(@account, @username)
+      end
+
+      def role
+        @role ||= @resource_class[role_id]
+        raise SecurityErr::RoleNotFound(role_id) if @username.nil?
+        return @role
+      end
+
+      # TODO add parsing format to check xms mirid is in proper format
```

Comment:
This is quite a big todo. I suggest that you create a method for `validate_xms_mirid_format` and leave it empty so it's more visible that it's missing.

Reply:
I don't think this is a big TODO, b/c as discussed we assuming that the xms_mirid claim has a consistent length and are verifying it is in the proper format in terms of number of  `/`

Something along the lines of :
```
def validate_xms_mirid_format
   raise Err:Error unless xms_mirid_hash.length == 4`
end
```

Reply:
Don't you want to verify that it consists of all the expected fields too?

Reply:
`@xms_mirid_hash.length == 4 && @xms_mirid_hash.key?("subscriptions" && "resourcegroups" && "providers")`. Would I need to check more IYO?

Additionally, would providers be `Microsoft.Compute/virtualMachines/sigalVMUbuntu` ? b.c rn I am splitting at the `/` and therefore breaking apart the the above string

Reply:
If we want to be `providers => Microsoft.Compute/virtualMachines/VMUbuntu` we need to change the logic a bit in terms of comparisons b/w user/system-assigned-identities

Reply:
It's not critical IMO. `@xms_mirid_hash.length == 4 && @xms_mirid_hash.key?("subscriptions" && "resourcegroups" && "providers")` is fine.